lib/ably/AblyExceptions.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * Generic exception for Ably classes
+ */
+class AblyException extends Exception {
+    
+    public function __construct($message, $code = 0) {
+        parent::__construct($message, $code);
+    }
+    
+}
+
+
+/**
+ * Exception thrown when a request to Ably API fails (HTTP response other than 200 or 201)
+ */
+class AblyRequestException extends AblyException {
+
+	private $response;
+    
+    public function __construct($message, $code, $response) {
```

Comment:
FYI, an AblyException typically has both a status code and a code.  The status code often maps to a normal HTTP status code, whereas the code is a unique Ably error code that we try and share across all libraries, see https://github.com/ably/ably-common/blob/master/protocol/errors.json.  See the [Java def of a AblyException](https://github.com/ably/ably-java/blob/master/src/io/ably/types/AblyException.java#L37).  I would try and follow the same naming convention if pssible.